src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -25,12 +26,13 @@ const GI_B: u64 = 1024 * MI_B;
 // All structs implement a simple ToQemuParams interface which allows their
 // user to convert them to actual qemu command line parameter strings.
 
-trait ToQemuParams {
+#[async_trait]
+trait ToQemuParams: Send + Sync {
     // OsString could look as a better fit here, however since foreign strings
     // come to this code from the outside as Strings already and this code adds
     // nothing but UTF-8 (in fact probably just ASCII) switching to OsStrings
     // now seems pointless.
-    fn qemu_params(&self) -> Result<Vec<String>>;
+    async fn qemu_params(&self) -> Result<Vec<String>>;
```

Comment:
This looks very wrong, too.  Can you elaborate why this function should be `async`?